celery/app/base.py

```diff
@@ -731,15 +731,26 @@ def send_task(self, name, args=None, kwargs=None, countdown=None,
                     root_id = parent.request.root_id or parent.request.id
                 if not parent_id:
                     parent_id = parent.request.id
-
-        message = amqp.create_task_message(
-            task_id, name, args, kwargs, countdown, eta, group_id,
-            expires, retries, chord,
-            maybe_list(link), maybe_list(link_error),
-            reply_to or self.oid, time_limit, soft_time_limit,
-            self.conf.task_send_sent_event,
-            root_id, parent_id, shadow, chain,
```

Comment:
`AMQP.create_task_message` attempts to be protocol-agnostic, so I am not sure that we should apply a conditional based on protocol version. I would suggest to add an additional variadic keyworded-argument list in the signature of [AMQP.as_task_v1](https://github.com/celery/celery/blob/master/celery/app/amqp.py#L401):

```
def as_task_v1(..., **compat_kwargs):
```

Then both signatures will be able to receive the keyworded arguments `kwargsrepr` & `argsrepr`, but only `as_task_v2` will be actually using them.

What do you think?

Reply:
Excellent point - for some reason I thought the `as_task` methods were coming from another library. I agree that it makes more sense to update those and remove the conditional. Updating now...